package.json

```diff
@@ -19,47 +19,72 @@
   },
   "license": "MIT",
   "engines": {
-    "node": ">=10.0.0",
+    "node": "^10.13 || ^12.13",
     "yarn": "^1.15.2"
   },
   "repository": {
     "type": "git",
     "url": "https://github.com/axa-group/oauth2-mock-server.git"
   },
-  "main": "index.js",
+  "main": "./dist/index.js",
+  "types": "./dist/index.d.ts",
   "bin": {
-    "oauth2-mock-server": "./bin/oauth2-mock-server.js"
+    "oauth2-mock-server": "./dist/oauth2-mock-server.js"
   },
   "files": [
     "CHANGELOG.md",
+    "LICENSE.md",
     "README.md",
-    "index.js",
-    "bin/",
-    "lib/"
+    "dist/**/*.*"
   ],
   "scripts": {
+    "build:clean": "rimraf ./dist",
+    "prebuild": "yarn build:clean",
+    "build": "tsc -p ./tsconfig.build.json",
     "cleanup:testresults": "rimraf TestResults",
-    "eslint": "eslint .",
-    "pretest": "yarn cleanup:testresults && yarn eslint",
+    "prelint": "tsc --noEmit",
+    "lint": "eslint --cache --cache-location .cache/ --ext=.js,.ts src test --max-warnings 0",
+    "pretest": "yarn cleanup:testresults && yarn lint",
     "test": "yarn jest"
   },
   "dependencies": {
+    "@types/node-jose": "^1.1.5",
```

Comment:
As our public type definitions depends on node-jose ones (`JWKKey`), this will automatically install the node-jose types upon installation of our package.

This makes me a little uneasy (as the types are not maintained in the `node-jose` repository), but I don't see currently any easy way to workaround this (one option would be to describe our own key type, but this would require some rewrite (which I consider out of the scope of this PR)). 

Would the `node-jose` type definition become haywire (namespace change, or similar) and the project referencing `oauth2-mock-server` also reference `@types/node-jose`, we would end up with a potential dependency hell.